src/avt/Database/CMakeLists.txt

```diff
@@ -8,99 +8,55 @@
 #  Cyrus Harrison, Thu Apr 19 16:07:11 PDT 2012
 #  Added Ghost/avtLocalStructuredDomainBoundaries.C
 #
+#  Kathleen Biagas, Tue Dec 13, 2022
+#  Moved setting of sources to subdirs.
+#
 #****************************************************************************
 
-SET(DATABASE_SOURCES
-Database/avtDatabase.C
-Database/avtDatabaseFactory.C
-Database/avtDatasetCollection.C
-Database/avtDatasetDatabase.C
-Database/avtFileDescriptorManager.C
-Database/avtGenericDatabase.C
-Database/avtIOInformation.C
-Database/avtSILGenerator.C
-Database/avtSourceFromDatabase.C
-Database/avtVariableCache.C
-Database/avtTransformManager.C
-Database/avtPolygonToTrianglesTesselator.C
-Database/avtPolyhedralSplit.C
-)
-
-SET(FORMATS_SOURCES
-Formats/avtFileFormat.C
-Formats/avtFileFormatInterface.C
-Formats/avtMTSDFileFormat.C
-Formats/avtMTSDFileFormatInterface.C
-Formats/avtMTMDFileFormat.C
-Formats/avtMTMDFileFormatInterface.C
-Formats/avtSTMDFileFormat.C
-Formats/avtSTMDFileFormatInterface.C
-Formats/avtSTSDFileFormat.C
-Formats/avtSTSDFileFormatInterface.C
-)
-
-SET(GHOST_SOURCES
-Ghost/StructuredBoundary.C
-Ghost/avtDomainBoundaries.C
-Ghost/avtGhostNodeGenerator.C
-Ghost/avtIsenburgSGG.C
-Ghost/avtStreamingGhostGenerator.C
-Ghost/avtStructuredDomainBoundaries.C
-Ghost/avtLocalStructuredDomainBoundaries.C
-Ghost/avtUnstructuredDomainBoundaries.C
-Ghost/avtNekDomainBoundaries.C
-Ghost/avtStructuredDomainNesting.C
-Ghost/avtUnstructuredPointBoundaries.C
-)
-
-SET(AVTDATABASE_SOURCES
-${DATABASE_SOURCES}
-${FORMATS_SOURCES}
-${GHOST_SOURCES}
-)
-
-INCLUDE_DIRECTORIES(
-${VISIT_COMMON_INCLUDES}
-${CMAKE_CURRENT_SOURCE_DIR}/Database
-${CMAKE_CURRENT_SOURCE_DIR}/Formats
-${CMAKE_CURRENT_SOURCE_DIR}/Ghost
-${VISIT_SOURCE_DIR}/avt/Math
-${VISIT_SOURCE_DIR}/avt/VisWindow/VisWindow
-${VISIT_SOURCE_DIR}/avt/DBAtts/MetaData
-${VISIT_SOURCE_DIR}/avt/DBAtts/SIL
-${VISIT_SOURCE_DIR}/avt/MIR/Base
-${VISIT_SOURCE_DIR}/avt/MIR/Tet
-${VISIT_SOURCE_DIR}/avt/MIR/Youngs
-${VISIT_SOURCE_DIR}/avt/MIR/Zoo
-${VISIT_SOURCE_DIR}/avt/MIR/Discrete/
-${VISIT_SOURCE_DIR}/avt/Pipeline/AbstractFilters
-${VISIT_SOURCE_DIR}/avt/Pipeline/Data
-${VISIT_SOURCE_DIR}/avt/Pipeline/Pipeline
-${VISIT_SOURCE_DIR}/avt/Pipeline/Sinks
-${VISIT_SOURCE_DIR}/avt/Pipeline/Sources
-${VISIT_SOURCE_DIR}/third_party_builtin/tess2
-${VISIT_SOURCE_DIR}/visit_vtk/full
-${VISIT_SOURCE_DIR}/visit_vtk/lightweight
-${VTK_INCLUDE_DIRS}
-)
-
-# Add link directories
-LINK_DIRECTORIES(${LIBRARY_OUTPUT_DIRECTORY})
+add_library(avtdatabase_ser)
```

Comment:
this looks nice! I would add a comment to these libs that states the sources are added via cmake logic that `add_subdirectory` pulls in, just to make it clear here where the heavy lifting is happening.

Reply:
I could add a section called 'CMake conventions' to our style guide, would that suffice?

Reply:
Yes, that sounds like a better place to describe the strategy.

Reply:
I added ticket #18426 to keep track of documenting.